geth/node/manager_test.go

```diff
@@ -321,6 +322,64 @@ func (s *ManagerTestSuite) TestReferences() {
 	}
 }
 
+// TestNodeSynchronizationFailure validates the failure of node synchronization if LightEthereum is disabled.
+func (s *ManagerTestSuite) TestNodeSynchronizationFailure() {
+	require := s.Require()
+
+	nodeConfig, err := MakeTestNodeConfig(params.RopstenNetworkID)
+	require.NoError(err)
+
+	nodeConfig.LightEthConfig.Enabled = true
+
+	nodeStarted, err := s.NodeManager.StartNode(nodeConfig)
+	require.NoError(err)
+	require.NotNil(nodeStarted)
+
+	defer s.NodeManager.StopNode()
+	<-nodeStarted
+
+	ethClient, err := s.NodeManager.LightEthereumService()
+	require.NoError(err)
+	require.NotNil(ethClient)
+
+	sync := node.NewSyncPoll(ethClient)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Millisecond)
+	defer cancel()
+
+	// Validate that synchronization failed because of time.
+	syncError := sync.Poll(ctx)
+	require.NotNil(syncError)
```

Comment:
It's possible to verify if the error is correct.  I think it should be equal to`context. DeadlineExceeded`.

Reply:
Oh actually, it should be a specific error from `syncpoll.go` file.